### PR TITLE
Hello Rails Front End: Create a new React app with Vite

### PR DESCRIPTION
**Pull request by danifromecuador**
## In this PR, I've:
- Created a `dev` and a `setup` branch and followed GitFlow pattern
- Created a new React app using Vite
- Created the store and slice
- Render the state of the store

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,165 @@
...
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 Hello Rails Back-End<a name="about-project"></a>
+
+**Hello Rails Back-End** is my first attempt to integrate backend and frontend using two separated projects, one for the backend and one for the frontend. The backend will provide an endpoint of a random greeting and the frontend will display it using React Redux for it.
+
+Link to the back end repository: [front-end](https://github.com/danifromecuador/hello-rails-backend/pull/1)
```
Comment (AmaduKamara) [inline README.md]: - [x] Kindly modify the title of this link to link to the back-end and not front-end to help prevent confusion 👍🏽

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      .catch((error) => {
+        console.error('Error fetching data:', error);
+      });
+  }, [dispatch]);
+
+  return (
+    <div className="App">
+      <Message />
+    </div>
+  );
```
Comment (AmaduKamara) [inline src/App.jsx]: - [x] Good job so far, but I will encourage you to have a route set up in this file that routes to the Greeting component or Message component in your case, using react-router-dom. It is required that you make use of react-router-dom for routing. Kindly add that 👍🏽

Screenshot 2023-11-02 at 1 55 51 AM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @danifromecuador,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You set up Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Fetching data from the backend correctly 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly add react-router package to handle routing to the Greeting or Message component as required in the project requirement. 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @danifromecuador 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes from the previous reviewer

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**